runner/src/rpc/api.rs

```diff
@@ -119,6 +129,54 @@ pub async fn view(
     Ok(serde_call_result)
 }
 
+pub async fn view_state(
+    contract_id: AccountId,
+    prefix: Option<StoreKey>,
+) -> Result<HashMap<String, Vec<u8>>, String> {
+    let query_resp = tool::json_client()
+        .call(&methods::query::RpcQueryRequest {
+            block_reference: BlockReference::Finality(Finality::Final),
+            request: QueryRequest::ViewState {
+                account_id: contract_id,
+                prefix: prefix.unwrap_or_else(|| vec![].into()),
+            },
+        })
+        .await
+        .map_err(|err| format!("Failed to query state: {:?}", err))?;
+
+    match query_resp.kind {
+        QueryResponseKind::ViewState(state) => Ok(tool::into_state_map(state.values)),
+        _ => Err("Could not retrieve access key".to_owned()),
+    }
+}
+
+pub async fn patch_state<T>(
+    account_id: AccountId,
+    key: String,
+    value: T,
```

Comment:
```suggestion
    value: &T,
```

This just needs to be serialized, should just pass references